Add init_batch_args to SampleQC stage query_command() and Refactor Reusability Checks

michael-harper opened the pull request:
This PR refactors allows additional customisation and initialisation options for the SampleQC batch process to provide more worker and driver memory as well as more driver cores.

This is done via providing `init_batch_args` to `query_command()`.

Alexander-Stuckey reviewed (approved):
Looks good to me

MattWellie commented on cpg_workflows/large_cohort/sample_qc.py:
```diff
@@ -15,7 +15,27 @@
 from gnomad.sample_qc.pipeline import annotate_sex
 
 
+def can_reuse_dataset(path: str):
+    """
+    checks for existence of a Path
+    if the path is a MT or VDS, checks for the success file
+    """
+
+    if not path:
+        return False
+
+    path_as_path = to_path(path)
+
+    if path_as_path.suffix in ['.mt', '.ht']:
+        path_as_path /= '_SUCCESS'
+    if path_as_path.suffix in ['.vds']:
+        path_as_path /= 'variant_data/_SUCCESS'
+
+    return path_as_path.exists()
```
this should be reused from the cpg_workflows module

MattWellie replied:
```
from cpg_workflows.utils import can_reuse
```

MattWellie replied:
can_reuse calls exists, exists calls exists_not_cached, exists_not_cached does this `_SUCCESS` check
https://github.com/populationgenomics/production-pipelines/blob/d37ca03a06bb9b8c0df22ea8ac13d3ed1d5de797/cpg_workflows/utils.py#L181

michael-harper commented:
Thanks @MattWellie ! I've removed the `can_resuse_dataset` function and gone back to the imported `can_reuse`. I added a logging statement that tells use if it's reusing the checkpointed vds.

Alexander-Stuckey commented on cpg_workflows/large_cohort/sample_qc.py:
```diff
@@ -16,6 +16,7 @@
 
 def run(vds_path: str, out_sample_qc_ht_path: str, tmp_prefix: str):
+    logging.basicConfig(level=logging.INFO)
     if can_reuse(out_sample_qc_ht_path):
         return []
```
If we can reuse the output, shouldn't we return the path?

Alexander-Stuckey reviewed (commented):
Just one comment

MattWellie reviewed (approved):
LGTMatt
